docs/DeveloperGuide.md

```diff
@@ -1,38 +1,392 @@
+
+Step 2. The user executes the `taken CS2113 y/1 s/2 g/A+` command to plan the module CS2113 for year 1 and semester 2 to be added into the list. The `taken` command is parsed through `Parser` which returns a `TakenCommand()` object if a valid command is provided. The `PlanCommand()` constructs a `Module()` object with the overloaded constructor `Module()`. It is instantiated with the respective arguments which sets the `isTaken` parameter `true`, `moduleCode`, `year`, `semester` and `grade`.
+
+Step 3 - 6. Identical to that of a `plan` command as mentioned above.
+
+_Design considerations:_
+**Aspect: How plan/taken executes:**
+- **Alternative 1 (current choice):** Have an overloaded Module() constructor which accepts both types of modules. The difference being the isTaken parameter.
+  - Pros: Easy to implement
+  - Cons: Less readability and harder to differentiate the two.
+- **Alternative 2:** Have two classes, TakenModule and PlanModule, for the 2 types.
+  - Pros: More readability and easier differentiation.
+  - Cons: More complex. 2 classes inherited + checking the class type in queries
+
+<br>
+
+### Remove module
+The Remove Module feature allows users to remove a module from the ModuleList using the command `remove`. It is facilitated by `LogicManager` and extends the abstract class `Command`. The module code is given as the argument to remove that specific module.
+
+Given below is an example scenario of the remove command and how it behaves at each step.
+
+Step 1. The user executes the `remove CS2113` command, given that a module `CS2113` exists within the `ModuleList`, to remove the module. The `remove` command is parsed through `Parser` which returns a `RemoveCommand()` object if a valid command is provided. The `RemoveCommand()` stores the string `moduleCode` as its attribute.
+
+Step 2. The `RemoveCommand` is then executed by the `LogicManager` calling `execute()`. Then, the `ModuleList`, which is retrieved from the `ModelManager` through `getModuleList()`, is iterated through to find the index of a `Module` with the same corresponding `moduleCode` through `getModule().getCode()`. This index is assigned to a variable.
+
+Step 3. It is then passed to the `ModelManager`, along with the index, and executes the `removeModule()` method on the `ModuleList` object. The `ModuleList` subsequently executes the `remove(index)` method to remove the module from the list.
+
+Step 4. Upon successful execution of the above, it is then passed back to `RemoveCommand` where  `CommandResult()` is constructed with the message to be printed to the user.
+
+Step 5. The `CommandResult` object is passed to the `Ui` component with a `printMessage()` method which prints the formatted message to the Command Line Interface.
+
+The following sequence diagram shows how the `remove` command works:
+![RemoveModuleSequenceDiagram](uml/diagrams/RemoveModSequence.png)
```

Comment:
<img width="579" alt="image" src="https://user-images.githubusercontent.com/46262670/229030532-12accf0d-d426-412c-8887-09427b16a066.png">

Here the lifelines shouldn't end with another boxes of class name